Remove rules converting cast varchar -> TO_CHAR() for oracle.

chrisknoll opened the pull request:
Fixes #97.

chrisknoll commented:
I'm looking into the failed tests now.

schuemie commented on inst/csv/replacementPatterns.csv:
```diff
@@ -18,20 +18,18 @@ oracle,"DATEADD(m,@months,@date)","ADD_MONTHS(@date, @months)"
-oracle,"DATEDIFF(day,@start, @end)",(CAST(@end AS DATE) - CAST(@start AS DATE))
-oracle,"DATEDIFF(dd,@start, @end)",(CAST(@end AS DATE) - CAST(@start AS DATE))
-oracle,"DATEDIFF(d,@start, @end)",(CAST(@end AS DATE) - CAST(@start AS DATE))
+oracle,"DATEDIFF(day,@start, @end)",(@end - @start)
```
Could you help me understand how this relates to the issue? The rules you changed did not use TO_CHAR. I'm pretty sure I introduced the casts to date for a valid reason (ie.  I think SQL Server auto-casts to DATE, but Oracle doesn't).

chrisknoll replied:
The rules were chaining together, as I was changing how CAST * as DATE) in impacted this rule.  It was at this point that I  realized that if you are doing a DATEDIFF(), whatever @start and @end resolves to should be a DATE type.  Any SqlRender rules will apply to whatever is inside @start and @end. So there is no CAST required on Oracle (we should assume the value in @start has already been properly resolved to a date), and all you do in oracle is subtract one from the other.

schuemie commented on inst/csv/replacementPatterns.csv:
```diff
@@ -18,20 +18,18 @@ oracle,"DATEADD(m,@months,@date)","ADD_MONTHS(@date, @months)"
 oracle,'@a' +,'@a' ||
 oracle,CAST(@a AS varchar(@b)) +,CAST(@a AS varchar(@b)) ||
 oracle,+ CAST(@a AS varchar(@b)),|| CAST(@a AS varchar(@b))
-oracle,CAST(@a AS varchar) +,CAST(@a AS varchar) ||
```
Similarly, I'm struggling to see how this relates to the issue. The rules here are specifically aimed at converting the '+' sign to a '||' token.

chrisknoll replied:
Cast(value as varchar) is not a valid syntax in Oracle, so this rule is invalid.  In oracle, all varchar must be followed by a length. 

In the case where we have a CAST(@a as Varchar) it makes sense to do the translated form as TO_CHAR(@a).  Note, you do not want to change the general cast(@a as varchar(#)) into TO_CHAR() because you loose the varchar length (which is the reason for the original issue and this PR).

chrisknoll commented:
@schuemie , sorry to rollback a bit, but the case that was originally found was related to select .. INTO statements:

```
translateSql("select cast('' as varchar(10)) as test into #test", targetDialect = "oracle")$sql
```
you get this:
``
 "select TO_CHAR('' ) as test into bydgk9vatest"
```

So, I removed the rule that drops the varchar length, and updated the tests that this behavior depended on (and rewrote a complex test that uses CONVERT(date...) instead of CAST(.. as date).
